File: app/src/main/java/de/kontranik/freebudget/model/RegularTransaction.java
package de.kontranik.freebudget.model;

import java.util.Date;

public class RegularTransaction {

    private Long id;
    private int month;
    private int day;
    private String description;
    private String category;
    private double amount;
    private long date_create;
    private long date_start;
    private long date_end;

    public RegularTransaction(long id) {
        this.id = id;
    }

    public RegularTransaction(int month, int day, String description, String category, double amount) {
        this.month = month;
        this.day = day;
        this.description = description;
        this.category = category;
        this.amount = amount;
        this.date_create = new Date().getTime();
        this.date_start = 0;
        this.date_end = 0;
    }

    public RegularTransaction(long id, int month, int day, String description, String category, double amount) {
        this.id = id;
        this.month = month;
        this.day = day;
        this.description = description;
        this.category = category;
        this.amount = amount;
        this.date_create = new Date().getTime();
        this.date_start = 0;
        this.date_end = 0;
    }

    public RegularTransaction(
            long id, int month, int day, String description, String category, double amount,
            long date_start, long date_end, long date_create) {
        this.id = id;
        this.month = month;
        this.day = day;
        this.description = description;
        this.category = category;
        this.amount = amount;
        this.date_create = date_create;
        this.date_start = date_start;
        this.date_end = date_end;
    }

    public RegularTransaction(int month, int day, String description, String category, double amount, long date_create) {
        this.month = month;
        this.day = day;
        this.description = description;
        this.category = category;
        this.amount = amount;
        this.date_create = date_create;
        this.date_start = 0;
        this.date_end = 0;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public int getMonth() {
        return month;
    }

    public void setMonth(int month) {
        this.month = month;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public double getAmount() {
        return amount;
    }

    public void setAmount(double amount) {
        this.amount = amount;
    }

    public long getDate_create() {
        return date_create;
    }

    public void setDate_create(long date_create) {
        this.date_create = date_create;
    }

    public int getDay() {
        return day;
    }

    public void setDay(int day) {
        this.day = day;
    }

    public long getDate_start() {
        return date_start;
    }

    public void setDate_start(long date_start) {
        this.date_start = date_start;
    }

    public long getDate_end() {
        return date_end;
    }

    public void setDate_end(long date_end) {
        this.date_end = date_end;
    }
}

File: app/src/main/java/de/kontranik/freebudget/model/RegularTransaction.kt
package de.kontranik.freebudget.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import de.kontranik.freebudget.database.DatabaseHelper
import java.util.*

@Entity(tableName = DatabaseHelper.TABLE_REGULAR)
class RegularTransaction {
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = DatabaseHelper.COLUMN_ID) var id: Long? = null
    @ColumnInfo(name = DatabaseHelper.COLUMN_MONTH) var month = 0
    @ColumnInfo(name = DatabaseHelper.COLUMN_DAY) var day = 0
    @ColumnInfo(name = DatabaseHelper.COLUMN_DESCRIPTION) var description: String = ""
    @ColumnInfo(name = DatabaseHelper.COLUMN_NOTE) var note: String? = null
    @ColumnInfo(name = DatabaseHelper.COLUMN_CATEGORY_NAME) var category: String = ""
    @ColumnInfo(name = DatabaseHelper.COLUMN_AMOUNT) var amount = 0.0
    @ColumnInfo(name = DatabaseHelper.COLUMN_DATE_CREATE) var dateCreate: Long = System.nanoTime()
    @ColumnInfo(name = DatabaseHelper.COLUMN_DATE_START) var dateStart: Long? = null
    @ColumnInfo(name = DatabaseHelper.COLUMN_DATE_END) var dateEnd: Long? = null

    constructor() {}

    @Ignore
    constructor(id: Long) {
        this.id = id
    }
    @Ignore
    constructor(month: Int, day: Int, description: String, category: String, amount: Double, note: String?) {
        this.month = month
        this.day = day
        this.description = description
        this.category = category
        this.amount = amount
        dateCreate = System.nanoTime()
        dateStart = null
        dateEnd = null
        this.note = note
    }
    @Ignore
    constructor(
        id: Long?,
        month: Int,
        day: Int,
        description: String,
        category: String,
        amount: Double,
        note: String?
    ) {
        this.id = id
        this.month = month
        this.day = day
        this.description = description
        this.category = category
        this.amount = amount
        dateCreate = System.nanoTime()
        dateStart = null
        dateEnd = null
        this.note = note
    }
    @Ignore
    constructor(
        id: Long, month: Int, day: Int, description: String, category: String, amount: Double,
        date_start: Long?, date_end: Long?, date_create: Long, note: String?
    ) {
        this.id = id
        this.month = month
        this.day = day
        this.description = description
        this.category = category
        this.amount = amount
        this.dateCreate = date_create
        this.dateStart = date_start
        this.dateEnd = date_end
        this.note = note
    }
    @Ignore
    constructor(
        month: Int,
        day: Int,
        description: String,
        category: String,
        amount: Double,
        date_create: Long,
        note: String?
    ) {
        this.month = month
        this.day = day
        this.description = description
        this.category = category
        this.amount = amount
        this.dateCreate = date_create
        dateStart = null
        dateEnd = null
        this.note = note
    }
}